### ISSUE TITLE
Null value confuses mapping, value assigned to wrong property

### ISSUE THREAD

**Issue by devnull73** [nhibernate/nhibernate-core#1399]
Hibernate noob, so please bear with me :)

I have a simple mapping, read-only, getting values from an oracle view:

```
  <class name="ext_scada_eoi_mw" table="ext_scada_eoi_mw_all_v" mutable="false">
    <cache usage="read-only"/>
    <composite-id>
      <key-property name="company_name" column="company_name" />
      <key-property name="trading_day" column="trading_day" type="Date" />
      <key-property name="trading_period" column="trading_period" />
      <key-property name="participant_name" column="participant_name" />
      <key-property name="resource_name" column="resource_name" />
    </composite-id>

...
 
    <property name="provisional_quantity_mw"  />
    <property name="final_quantity_mw"  />
</class>
```

The last 2 properties are nullable, ie:
```
	public class ext_scada_eoi_mw
	{
...
		[DataMember(IsRequired = false, Order = 30)]
		public virtual Decimal? provisional_quantity_mw { get; set; }
		[DataMember(IsRequired = false, Order = 31)]
		public virtual Decimal? final_quantity_mw { get; set; }
...
	}
```

However, when one of the values contains a NULL, it actually assigns the value to the wrong property. The SQL generated is 100% ok.

```
SELECT this_.company_name as compa1_18_0_, this_.trading_day as tradi2_18_0_, this_.trading_period as tradi3_18_0_, this_.participant_name as parti4_18_0_, this_.resource_name as resou5_18_0_, this_.trading_period_start as tradi6_18_0_, 
...
this_.provisional_quantity_mw as prov30_18_0_, this_.final_quantity_mw as fina31_18_0_ 
FROM ext_scada_eoi_mw_all_v this_ 
WHERE this_.company_name = :p0 and this_.trading_day = :p1;
```

The data returned by this query has a NULL for provisional_quantity_mw, and a value for final_quantity_mw, as expected.

However, when returning the data via my service (or even via a simple TestMethod) I get:
```
            "provisional_quantity_mw": 183.36,
            "final_quantity_mw": null
```

In other words, it looks like the null value for provisional_quantity_mw failed, it then picked up the next value, and final_quantity_mw gets a default null value.... or something like that.

Any help would be appreciated!

**Comment by fredericDelaporte** [comment]
This place here is for reporting proven bugs or feature requests. Please ask for support on [nhusers group](http://groups.google.com/group/nhusers) or on some other places like StackOverflow.

If you are sure there is a bug on NHibernate side, please elaborate a test case. See [contributing](https://github.com/nhibernate/nhibernate-core/blob/master/CONTRIBUTING.md) for instructions on that subject.

**Comment by devnull73** [comment]
Im glad you closed it, because I found the embarrassing solution to my issue this morning.

Im thinking of a new career at this point ;)
